src/chunk.c

```diff
@@ -41,12 +47,184 @@ chunk_create_from_tuple(HeapTuple tuple, int16 num_constraints)
 	return chunk;
 }
 
+/*
+ * Align a chunk's hypercube in 'aligned' dimensions.
+ *
+ * After a chunk collision scan, this function is called for each chunk in the
+ * chunk scan context. If a chunk includes slices for an aligned dimension, we
+ * cut the context's hypercube to not overlap with that slice.
+ */
+static bool
```

Comment:
I don't understand why we are cutting aligned dimensions first. Shouldn't we cut them last. Actually shouldn't it be an error if we need to cut an aligned dimension?

Reply:
If you think as time as an aligned dimension need for deleting evenly across space partitions it seems that it would be dangerous to cut it